otherlibs/runtime_events/runtime_events.ml

```diff
@@ -156,25 +156,108 @@ module Timestamp = struct
     t
 end
 
+module Type = struct
+
+  (* the data structure is primarily managed in C *)
+  type[@warning "-unused-field"] 'a custom = {
+    serialize: 'a -> bytes;
```

Comment:
For the sake of efficiency, is it conceivable that the size of custom types be limited (it'll be limited by the actual ring size anyway) to, for example, 4kB?

Then one can replace this by:

```ocaml
serialize: bytes -> 'a -> int;
```

where the serializer is passed a pre-allocated 4kB buffer that is reused for every event. This could save many allocations.

Reply:
Indeed events have a limited length (2K), it definitely makes sense to use a pre-allocated buffer.


For deserialization the same can be achieved by providing a view of the buffer: `deserialize: bytes -> int -> 'a`, the second argument being the size of the data.